ci/drone/functions.star

```diff
@@ -34,7 +34,7 @@ def unix_common(install_script, buildscript_to_run):
         # Default build script (if not exists) 
         # 'if [ ! -e .drone/drone.sh ]; then %s; fi' % download_from_boostCI('drone.sh', '.drone'),
         # Chosen build script inside .drone (if a filename and does not exist)
-        # 'if [ "$(basename "{0}")" = "{0}" ] && [ ! -e .drone/{0} ]; then {1}; fi'.format(buildscript_to_run, download_from_boostCI(buildscript_to_run, '.drone')),
+        'if [ "$(basename "{0}")" = "{0}" ] && [ ! -e .drone/{0} ]; then {1}; fi'.format(buildscript_to_run, download_from_boostCI(buildscript_to_run, '.drone')),
```

Comment:
regarding this conditional:
```
'if [ "$(basename "{0}")" = "{0}" ]
```
When would that check fail? 
In the case it fails would there be any harm in attempting curl anyway?
If we could remove that conditional it would make functions.star smaller and it cleans up drone output log files which users see.

Also, trying to figure out if there could be any way to remove this line
```
'if [ "$(basename "$DRONE_REPO")" != "boost-ci" ]; then',
```
which will appear in every job, every repository, every log file, yet it  only optimizes a case for one repository. Could that conditional be moved inside "download_from_boostCI"?

Reply:
> When would that check fail?

I wanted to be extra careful and skip the download when the script was something like "../mydir/script.sh"

> Also, trying to figure out if there could be any way to remove this line

Though the same: Although the download function had a 2nd parameter we didn't use it. I.e. we always download to the same location the file is on BoostCI. Hence I moved the "exists"-check into the download function and removed the check for BoostCI: When testing BoostCI the file will always be there so the exist check will succeed and now download will happen